### PR TITLE
Add unit tests

### PR DESCRIPTION
**Pull request by miguelmtzinf**

### CODE REVIEW

File: test/BridgeExecutorBase.spec.ts [hunk 1]
```diff
@@ -399,6 +400,46 @@ describe('BridgeExecutorBase', async function () {
       expect((await bridgeExecutor.getActionsSetById(0)).executed).to.be.equal(true);
     });
 
+    it('Queue and fail to execute an actions set with insufficient value', async () => {
```
Comment (miguelmtzinf) [inline test/BridgeExecutorBase.spec.ts:403]: Following the style of the tests, the title would be "Tries to queue.... value (revert expected)'

File: test/BridgeExecutorBase.spec.ts [hunk 2]
```diff
@@ -399,6 +400,46 @@ describe('BridgeExecutorBase', async function () {
...
+
+      const { data } = encodeSimpleActionsSet(greeter.address, 'setMessage(string)', [NEW_MESSAGE]);
+      const value = (await ethers.provider.getBalance(bridgeExecutor.address)).add(1);
+      await expect(bridgeExecutor.queue(
+        data[0] as string[],
+        [value],
+        data[2] as string[],
+        data[3] as string[],
+        data[4] as boolean[]
+      )).to.not.be.reverted;
```
Comment (miguelmtzinf) [inline test/BridgeExecutorBase.spec.ts:415]: Here is the only one use of `to.not.be.reverted`.
The others tests just do `expect(await tx)` , and additionally check for event emissions

File: test/BridgeExecutorBase.spec.ts [hunk 3]
```diff
@@ -399,6 +400,46 @@ describe('BridgeExecutorBase', async function () {
...
+        ExecutorErrors.InsufficientBalance
+      );
+    });
+
+    it('Queue and execute an actions set to self-destruct via delegatecall', async () => {
+      const oops = await new Oops__factory(user).deploy();
+      const data = oops.interface.encodeFunctionData('oops');
+
+      await expect(bridgeExecutor.queue([oops.address], [0], [''], [data], [true])).to.not.be
+        .reverted;
```
Comment (miguelmtzinf) [inline test/BridgeExecutorBase.spec.ts:431]: same as above

File: test/BridgeExecutorBase.spec.ts (hunk 1 above)
Comment (Zer0dot) [reply test/BridgeExecutorBase.spec.ts:403]: Sure, will fix.

Comment (Zer0dot) [reply test/BridgeExecutorBase.spec.ts]: I fixed this but I don't like this convention long-term. I think we should separate revert cases from success cases in separate sections instead. This naming convention can get a little confusing if the test contains multiple transactions (which, admittedly isn't often, but still).

File: test/BridgeExecutorBase.spec.ts (hunk 2 above)
Comment (miguelmtzinf) [reply test/BridgeExecutorBase.spec.ts:415]: Agree

File: test/BridgeExecutorBase.spec.ts (hunk 1 above)
Comment (miguelmtzinf) [reply test/BridgeExecutorBase.spec.ts]: Yeah, thats a good point actually

**Pull request merged**